src/protobuf/bmv2/table.proto

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "google/protobuf/struct.proto";
+
+import "src/protobuf/bmv2/primitives.proto";
+
+// A pipeline is a sequence of actions and table applications
+// defined within the scope of one "control" construct
+// usually there are two pipelines in a p4 program
+// one for ingress and one for egress
+message Pipeline {
+  // name assigned by programmer
+  // e.g. MyIngressPipeline
+  string name = 1;
+  // Auto-generated id
+  int32 id = 2;
+  // Source information about where the pipeline is defined
+  SourceInfo source_info = 3;
+  // The first table matched on
+  string init_table = 4;
+  // All the tables and their actions that are part of this pipeline
+  repeated Table tables = 5;
+  // TODO(babman): action_profiles
+  // TODO(babman): conditionals
+}
+
+// A control plane table definition
+// including its keys and actions
+message Table {
+  // Full qualified name assigned by programmer
+  // e.g. MyIngressPipeline.MyTable
+  string name = 1;
+  // Auto-generated id
```

Comment:
Is this really autogenerated? I think this is the P4Runtime table ID, no? It can be generated, but you can also pick it as a programmer (see https://s3-us-west-2.amazonaws.com/p4runtime/docs/master/P4Runtime-Spec.html#sec-id-allocation). It would be good to clarify the comment, here and for other IDs)